src/object.d

```diff
@@ -3190,11 +3199,24 @@ void __ctfeWriteln(T...)(auto ref T values) { __ctfeWrite(values, "\n"); }
  * Create RTInfo for type T
  */
 
+template RTInfoImpl(size_t[] pointers)
+{
+    immutable size_t[pointers.length] data = pointers[];
+    immutable RTInfoImpl = data.ptr;
```

Comment:
> RTInfo is already a pointer into the buffer, and the compiler/linker is free to omit the (length,ptr) pair. Or do you have another indirection in mind?

Ah right, that means you get the length from the type itself or from the typeinfo?
In the later case it would be another random memory access.
If we already have the length, then of course we don't need to store it.

> 16 bytes. The 10 bytes are likely aligned to 16 bytes anyway.

The alignment for ubyte* is only a single byte, so we can really squash this. Section alignment is only done when merging sections of different object files, and even then `--sort-section=alignment` is a simple linker optimization.

> This could be a nice optimization, but we loose the easy accessible information of the size of a struct that is repeated in an array.

We shouldn't optimize this before we have some stats.

> I think the more important part is to optimize the way the bitmap pointers are stored alongside the allocated memory:

Maybe we need to rework the weird BlkAttr.APPENDABLE implementation?
If we can leverage the information then it makes sense to move most array functionality from rt.lifetime into the GC.

> for large allocations we might just store the pointer to the RTInfo

Let's assume large objects with GC references are rare for now. Again empiric data would be nice, maybe someone can get them for larger projects like [higgsjs/Higgs](https://github.com/higgsjs/Higgs) and [Hackerpilot/DCD](https://github.com/Hackerpilot/DCD).